packages/sso-express/src/controllers.ts

```diff
@@ -3,6 +3,8 @@ import { BaseClient, generators, TokenSet } from "openid-client";
 import { getSessionRemainingTime, isAuthenticated } from "./helpers";
 import { SSOExpressOptions } from "./index";
 
+const code_verifier = generators.codeVerifier();
```

Comment:
You want to generate the `code_verifier` in the request handler, and save it in the session (see how `state` is created).

If you create the code verifier here:
 - it's long-lived, so an attacker has more time to be able to guess it
 - it's going to be different for each server replica, and your auth request and callback might be handled by different replicas (e.g. if using a round-robin load balancer)